pkg/wharfapi/token.go

```diff
@@ -1,150 +1,63 @@
 package wharfapi
 
 import (
-	"encoding/json"
 	"fmt"
-	"net/http"
-	"net/url"
+
+	"github.com/google/go-querystring/query"
+	"github.com/iver-wharf/wharf-api-client-go/v2/pkg/model/request"
+	"github.com/iver-wharf/wharf-api-client-go/v2/pkg/model/response"
 )
 
-// Token is a value used by the provider plugins to authenticate with the remote
-// providers.
-type Token struct {
-	TokenID  uint   `json:"tokenId"`
-	Token    string `json:"token"`
-	UserName string `json:"userName"`
+// TokenSearch is used when getting tokens without using a token ID
+// through the HTTP request:
+//  GET /api/token
+type TokenSearch struct {
+	Limit         *int     `url:"limit,omitempty"`
+	Offset        *int     `url:"offset,omitempty"`
+	OrderBy       []string `url:"orderby,omitempty"`
+	UserName      *string  `url:"userName,omitempty"`
+	UserNameMatch *string  `url:"userNameMatch,omitempty"`
 }
 
-// GetTokenByID fetches a token by ID by invoking the HTTP request:
-// 	GET /api/token/{tokenID}
-func (c Client) GetTokenByID(tokenID uint) (Token, error) {
-	newToken := Token{}
-
-	apiURL := fmt.Sprintf("%s/api/token/%v", c.APIURL, tokenID)
-	ioBody, err := doRequest("GET | TOKEN |", http.MethodGet, apiURL, []byte{}, c.AuthHeader)
```

Comment:
I'm not even sad we're loosing these endpoint names, ex `"GET | TOKEN |"`. Those logging messages were confusing to begin with and quite inconsistent.

Maybe we want to bring something like that back, but let's ignore that until we notice we want it.

Reply:
Agreed 👍🏻